pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -336,11 +336,39 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	// Remove dead container log symlinks.
 	// TODO(random-liu): Remove this after cluster logging supports CRI container log path.
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
-	for _, logSymlink := range logSymlinks {
-		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
-			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+	if cgc.manager.runtimeName == "docker" {
+		for _, logSymlink := range logSymlinks {
+			if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+				firstLinkDstPath, err := os.Readlink(logSymlink)
+				if err != nil {
+					glog.Errorf("Failed to resolve legacy container log symlink %q: %v", logSymlink, err)
+					continue
+				}
+				secondLinkDstPath, err := os.Readlink(firstLinkDstPath)
+				if err != nil {
```

Comment:
logSymlink should also be removed even if secondLink not exist

Reply:
Sir，as mentioned in  my first comment，this PR fix two problems as listed below:
1、Using "docker rm -f" to remove the container of pod(no matter whether the container is running or exited) will cause its logsymlink(softlink) residual;
2、Docker rotate container's log will cause the logsymlink unhealthy in an extremely short moment,during which the kube garbage collection will detect the unhealthy logsymlink and delete it.
Problems 2 is the exact reason why logSymlink should be kept when the secondLink is not exist(during docker logrotate).

Reply:
You're right, Good catch

Reply:
Many thanks @feiskyer